test/CMakeLists.txt

```diff
@@ -0,0 +1,93 @@
+cmake_policy(SET CMP0135 NEW)
+
+include(FetchContent)
+include(GoogleTest)
+
+set(BUILD_GMOCK OFF)
+set(INSTALL_GTEST OFF)
+set(gtest_force_shared_crt ON)
+
+FetchContent_Declare(
+    gtest
+    URL https://github.com/google/googletest/archive/main.tar.gz
+)
+FetchContent_MakeAvailable(gtest)
```

Comment:
Instead of unconditionally downloading and building GoogleTest from source, could you please try to look for the library on the build machine with `find_package(GTest)` and fall back to FetchContent only if not found?

Reply:
Yes, fallback option would be nice.

```cmake
find_package(GTest)

if(NOT GTest_FOUND)
    if(POLICY CMP0135)
        cmake_policy(SET CMP0135 NEW)
    endif()

    include(FetchContent)
    include(GoogleTest)

    set(BUILD_GMOCK OFF)
    set(INSTALL_GTEST OFF)
    set(gtest_force_shared_crt ON)

    FetchContent_Declare(
        gtest
        URL https://github.com/google/googletest/archive/main.tar.gz
    )
    FetchContent_MakeAvailable(gtest)
endif()
```

Reply:
Yeah this looks good

Reply:
It turns out that `FindGTest.cmake` will include the `GoogleTest` module also. So, `include(GoogleTest)` is actually redundant.

https://github.com/Kitware/CMake/blob/df52b85b0795404eec9e076ba4e6df7caaf40490/Modules/FindGTest.cmake#L102